### DISCUSSION TITLE
The invitation link for registering a new user is not sent to the new user's email

### DISCUSSION THREAD

**Discussion by nloshka404** [firefly-iii/firefly-iii#7965] [Firefly III]
The invitation link for registering a new user is not sent to the new user's email
The test email is sent directly.
Or should the invitation link not be sent to the new user's email?

**Comment by JC5** [comment]
Are there errors in your logs?

**Reply by nloshka404** [reply]
there are no errors in the log files
Or is there somewhere else I can look for errors?

**Reply by JC5** [reply]
At your provider perhaps? Or set MAIL_MAILER=log then the message will be saved in your logs instead of being sent out.

**Reply by nloshka404** [reply]
sending occurs via a third-party smtp server located on another host

**Reply by nloshka404** [reply]
# If you want Firefly III to mail you, update these settings
# For instructions, see: https://firefly-iii.readthedocs.io/en/latest/installation/mail.html
MAIL_DRIVER=smtp
MAIL_HOST=smtp.host.com
MAIL_PORT=465
MAIL_FROM=no-reply@**
MAIL_USERNAME=no-reply@**
MAIL_PASSWORD=Passwd
MAIL_ENCRYPTION=ssl
MAIL_MAILER=log
# Other mail drivers:
MAILGUN_DOMAIN=
MAILGUN_SECRET=
MANDRILL_SECRET=
SPARKPOST_SECRET=

**Reply by nloshka404** [reply]
Where to look for the sending log, what is the path to write this log file?

**Reply by nloshka404** [reply]
# Firefly III can send you the following messages
SEND_REGISTRATION_MAIL=true
SEND_ERROR_MESSAGE=true

**Reply by nloshka404** [reply]
I send an invitation and there is nothing in the mail and there are no entries in the logs locally
I have already deleted the link several times and sent it again several times in a row and there is nothing in the logs.
What am I doing wrong?

**Reply by nloshka404** [reply]
logs]# cat ff3-audit-2023-10-04.log 
[2023-10-04T15:31:58.512567+03:00] local.INFO: AUDIT: User visits admin index. (* (*@x512.ru) -> GET:https://b.x512.ru/admin) [] []
[2023-10-04T15:32:01.834704+03:00] local.INFO: AUDIT: User sends test message. (* (*@x512.ru) -> POST:https://b.x512.ru/admin/test-message) [] []
[2023-10-04T15:32:02.212489+03:00] local.INFO: AUDIT: User visits admin index. (* (*@x512.ru) -> GET:https://b.x512.ru/admin) [] []
[2023-10-04T15:37:56.483900+03:00] local.INFO: AUDIT: User sends test message. (* (*@x512.ru) -> POST:https://b.x512.ru/admin/test-message) [] []
[2023-10-04T15:37:56.608500+03:00] local.INFO: AUDIT: User visits admin index. (* (*@x512.ru) -> GET:https://b.x512.ru/admin) [] []
[2023-10-04T15:37:59.831896+03:00] local.INFO: AUDIT: User visits admin index. (* (*@x512.ru) -> GET:https://b.x512.ru/admin) [] []

**Reply by nloshka404** [reply]
../firefly-iii/config
Maybe something needs to be edited?

**Reply by JC5** [reply]
Do you also have the normal logs, not the audit logs?

**Reply by nloshka404** [reply]
logs]# ls
ff3-audit-2023-10-04.log
just this

#pwd
/var/www/b/firefly-iii/storage/logs

**Reply by nloshka404** [reply]
where can I see normal logs?
In the Nginx logs?
Tell me where to look and what to watch?

**Reply by JC5** [reply]
What are the settings for [LOG_CHANNEL](https://github.com/firefly-iii/firefly-iii/blob/main/.env.example#L46) and [APP_LOG_LEVEL](https://github.com/firefly-iii/firefly-iii/blob/main/.env.example#L52)?

If you can, set them to LOG_CHANNEL=stack, APP_LOG_LEVEL=debug

**Reply by nloshka404** [reply]
Should I send the logs to your email?

**Reply by nloshka404** [reply]
There are no error logs, everything is well written that what was sent, but there is nothing in the mail

**Comment by nloshka404** [comment]
Now the test letter does not arrive in the mail

**Comment by nloshka404** [comment]
Maybe the problem is in the mail server and not in the scripts? You tell me if the problem is not in the scripts, then I will torment the mail server why it refuses to accept and deliver mail.

**Reply by JC5** [reply]
As far as I can tell from your logs (my Russian is not that good) I see two or three test messages and then the two invites as I would expect. So yes, I believe your mail server is the one to be tormented. Looking at your settings, I would expect this:

```
#MAIL_DRIVER=smtp # REMOVE THIS LINE
MAIL_HOST=smtp.host.com
MAIL_PORT=465
MAIL_FROM=no-reply@**
MAIL_USERNAME=no-reply@**
MAIL_PASSWORD=Passwd
MAIL_ENCRYPTION=ssl
MAIL_MAILER=smtp
```

Perhaps change `MAIL_ENCRYPTION=ssl` to `MAIL_ENCRYPTION=tls`

**Reply by nloshka404** [reply]
After applying the recommendations you described above, everything worked.
Thanks a lot.
Have a great day!
